sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -182,6 +182,28 @@ properties. The implementation of the class defined here should preserve the
 information provided as compile-time constant properties through all uses of the
 pointer unless noted otherwise.
 
+.Not recommended Usage Example
+[source,c++]
+----
+using sycl::ext::oneapi::experimental;
+struct MyType {
+  MyType(const MyType& otherM) {
+     ...
+  }
+};
+
+struct MyKernel {
+  annotated_ptr<MyType, properties<PropA>> a; //non-trivially-copyable type
+  ...
+  void operator()() const {
+    ...
+  }
+};
+----
+It is not recommended to apply `annotated_ptr` to a non-trivially-copyable type `T`. After
+dereferencing an `annotated_ptr` to an `annotated_ref`, many operators in `annotated_ref` are
+return-by-value and pass-by-value, and thus might result in inefficient memory copies.
+
```

Comment:
I  think we should go further and say that `T` must be trivially copyable.  The semantics of many of the properties are unclear when `T` is not trivially copyable, and I think we have no use case for non-trivially copyable `T`.